Shopify/ruby-lsp
# Inhibit server restarts during bisect as well

### Motivation

This is a revival of what I wrote in https://github.com/Shopify/ruby-lsp/issues/2387#issuecomment-2288146864

I consistently run into the issue of the lockfile changing during a bisect, restarting the server which then runs bundler, which blocks `bundler/inline` from processing until the gems are installed which may or may not finish in a reasonable amount of time.

This stops the server from restarting when a bisect is in process, where the lockfile is likely to constantly change.

### Implementation

Piggy-back of the already existing code for checking against a rebase. This does still cause additional restarts if you bisect during a rebase or the other way around but it seems very rare and this implementation is very simple.

### Automated Tests

No. There are no tests for rebase

### Manual Tests

Start a bisect, and observe that the server doesn't restart itself. Similarly, with a rebase
